plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/lambda.ts

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Span, SpanKind, Tracer } from '@opentelemetry/api';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import {
+  AwsSdkInstrumentationConfig,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '../types';
+import { RequestMetadata, ServiceExtension } from './ServiceExtension';
+import { TextMapSetter, context, propagation } from '@opentelemetry/api';
+import type { ClientContext } from 'aws-lambda';
+
+export class LambdaServiceExtension implements ServiceExtension {
+  requestPreSpanHook(request: NormalizedRequest): RequestMetadata {
+    const functionName = this.extractFunctionName(request.commandInput);
+
+    let spanAttributes = {};
+    let spanName: string | undefined;
+
+    switch (request.commandName) {
+      case 'Invoke':
+        spanAttributes = {
+          [SemanticAttributes.FAAS_INVOKED_NAME]: functionName,
+          [SemanticAttributes.FAAS_INVOKED_PROVIDER]: 'aws',
+        };
+        if (request.region) {
+          spanAttributes = {
+            ...spanAttributes,
+            [SemanticAttributes.FAAS_INVOKED_REGION]: request.region,
+          };
+        }
+        spanName = `${functionName} invoke`;
+        break;
+    }
+    return {
+      isIncoming: false,
+      spanAttributes,
+      spanKind: SpanKind.CLIENT,
+      spanName,
+    };
+  }
+
+  requestPostSpanHook = (request: NormalizedRequest) => {
+    switch (request.commandName) {
+      case 'Invoke':
+        {
+          request.commandInput = injectPropagationContext(request.commandInput);
+        }
+        break;
+    }
+  };
+
+  responseHook(
+    response: NormalizedResponse,
+    span: Span,
+    tracer: Tracer,
+    config: AwsSdkInstrumentationConfig
+  ) {
+    const operation = response.request.commandName;
+
+    if (operation === 'Invoke') {
+      if (response.data && '$metadata' in response.data) {
+        span.setAttribute(
+          SemanticAttributes.FAAS_EXECUTION,
+          response.data['$metadata'].requestId
+        );
+      }
+    }
+  }
+
+  extractFunctionName = (commandInput: Record<string, any>): string => {
+    return commandInput?.FunctionName;
+  };
+}
+
+class ContextSetter implements TextMapSetter<Record<string, any>> {
+  set(carrier: Record<string, any>, key: string, value: string): void {
+    const parsedClientContext: ClientContext = JSON.parse(
+      carrier.ClientContext !== undefined
+        ? Buffer.from(carrier.ClientContext, 'base64').toString('utf8')
+        : '{"Custom":{}}'
+    );
+    const updatedPayload = {
+      ...parsedClientContext,
+      Custom: {
+        ...parsedClientContext.Custom,
+        [key]: value,
+      },
+    };
+    const encodedPayload = Buffer.from(JSON.stringify(updatedPayload)).toString(
+      'base64'
+    );
+    if (encodedPayload.length <= 3583) {
+      carrier.ClientContext = encodedPayload;
+    }
```

Comment:
Do we want to log anything to diag logger in case context cannot be injected?
Simialar to [this](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/760c4d4e9bd500ab65980f7414f72ad93da42e2f/plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/MessageAttributes.ts#L80)

Reply:
Added